http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/service/RuntimeServiceInterp.scala

```diff
@@ -224,25 +224,30 @@ class RuntimeServiceInterp[F[_]: Parallel](config: RuntimeServiceConfig,
       ctx <- as.ask
       paramMap <- F.fromEither(processListParameters(params))
       runtimes <- RuntimeServiceDbQueries.listRuntimes(paramMap._1, paramMap._2, googleProject).transaction
-      runtimesAndProjects = runtimes.map(r => (r.googleProject, r.samResource))
       _ <- ctx.span.traverse(s => F.delay(s.addAnnotation("DB | Done listRuntime db query")))
-      samVisibleRuntimes <- NonEmptyList.fromList(runtimesAndProjects).traverse { rs =>
+      runtimesAndProjects = runtimes.map(r => (r.googleProject, r.samResource))
+      samVisibleRuntimesOpt <- NonEmptyList.fromList(runtimesAndProjects).traverse { rs =>
         authProvider
           .filterUserVisibleWithProjectFallback(
             rs,
             userInfo
           )
       }
       _ <- ctx.span.traverse(s => F.delay(s.addAnnotation("Sam | Done visible runtimes")))
-    } yield {
-      // Making the assumption that users will always be able to access runtimes that they create
-      // Fix for https://github.com/DataBiosphere/leonardo/issues/821
-      runtimes
-        .filter(c =>
-          c.auditInfo.creator == userInfo.userEmail || samVisibleRuntimes.contains((c.googleProject, c.samResource))
```

Comment:
This is the bug: `samVisibleRuntimes` is a `Option[List[(GoogleProject, RuntimeSamResourceId)]]` so `.contains((proj, samResource))` always returns false